lib/AST/Module.cpp

```diff
@@ -426,8 +427,8 @@ void Module::lookupMember(SmallVectorImpl<ValueDecl*> &results,
       return VD->getModuleContext() == this;
     });
 
-    const DeclContext *accessScope = nominal->getFormalAccessScope();
-    if (accessScope && !accessScope->isModuleContext())
+    auto accessScope = nominal->getFormalAccessScope();
+    if (!accessScope.isPublic() && !accessScope.isModuleScopeContext())
```

Comment:
Ah, this changes meaning. "isModuleContext" = is a module = `internal`. "isModuleScopeContext" = is a module or a single file = `internal` or `fileprivate`. Really this is trying to ask "are we in a per-file context?".